### ISSUE TITLE
Responsive issue

### ISSUE THREAD

**Issue by selftraveler** [NUKnightLab/juxtapose#136]
Hi there, I love this so much, we are using it for a before and after article on home staging and it looks great on desktop.  However, on mobile there is a HUGE space between each juxtapose as I'm assuming it's still taking on the height assigned and not adjusting as needed.  I can't figure out how to fix it. Can someone please help me?  Thank you!!!  https://www.fosterdouglas.com/pro-tips-on-home-staging/

**Comment by selftraveler** [comment]
Fixed it with a post from in here!  Thank you!!!

**Comment by scott2b** [comment]
In general, the problem of a gap occurring under a Juxtapose embed is due to the iframe height being specified larger than the image rendering. Juxtapose will preserve the aspect ratio of the original images rather than try to fill that gap.

The fix for this is to use media queries to create a responsive iframe, setting the height of the iframe as needed for different viewport sizes.

If this problem persists for you, please submit a ticket at https://knightlab.zendesk.com

**Comment by selftraveler** [comment]
Hi there, I did use queries in the css, but had to add javascript for the queries to override the coding.


<div class="juxtapose-container">
  <div class="juxtapose">
    <img src="http://example.com/firstimage.jpg" data-label="First" />
    <img src="http://example.com/secondimage.jpg" data-label="Second" />
  </div>
</div>

<link rel="stylesheet" href="https://cdn.knightlab.com/libs/juxtapose/latest/css/juxtapose.css">
<script src="https://cdn.knightlab.com/libs/juxtapose/latest/js/juxtapose.min.js"></script>
<script>
var $juxtapose = $('.juxtapose'),
    $juxtapose_container = $('.juxtapose-container'),
    juxtapose_ratio;

$(window).load(function(){
  juxtapose_ratio = $juxtapose.outerHeight() / $juxtapose.outerWidth();
});

$(window).resize(function() {
  var new_width = $juxtapose_container.outerWidth(),
      new_height = new_width*juxtapose_ratio;

  $juxtapose.css({
    width: new_width,
    height: new_height
  })
});
</script>

**Comment by madhaze** [comment]
fix also documented on #105 

not sure why doesn't seem to be working when window id resized.
```
      var self = this;
      window.addEventListener("resize", function() {
        self.setWrapperDimensions();
      });
```
